src/components/EditContactSelector.tsx

```diff
@@ -0,0 +1,205 @@
+import {
+  DropdownMenu,
+  DropdownMenuContent,
+  DropdownMenuItem,
+  DropdownMenuPortal,
+  DropdownMenuTrigger,
+} from '@radix-ui/react-dropdown-menu';
+import { selectContact, type SelectContact } from '@src/server/db/models';
+import { type Dispatch, useEffect, useReducer } from 'react';
+import {
+  useFieldArray,
+  type Control,
+  type FieldErrors,
+  type UseFormRegister,
+} from 'react-hook-form';
+import { type z } from 'zod';
+import {
+  Discord,
+  Email,
+  Facebook,
+  Instagram,
+  Twitter,
+  Website,
+  Youtube,
+  type logoProps,
+} from './ContactIcons';
+import { type modifyDeletedAction } from '@src/pages/directory/[clubId]/edit/contacts';
+import { type editClubSchema } from '@src/utils/formSchemas';
+
+type Contact = Omit<SelectContact, 'clubId'>;
+
+function Reducer(
+  state: Array<Contact['platform']>,
+  action:
+    | {
+        type: 'add' | 'remove';
+        target: Contact['platform'];
+      }
+    | {
+        type: 'reset';
+        used: Contact['platform'][];
+        base: Contact['platform'][];
+      },
+) {
+  switch (action.type) {
+    case 'remove':
+      return state.filter((x) => x != action.target);
+    case 'add':
+      return [...state, action.target];
+    case 'reset':
+      return action.base.filter((x) => !action.used.includes(x));
+  }
+}
+const startContacts: Array<Contact['platform']> = [
+  'discord',
+  'instagram',
+  'website',
+  'email',
+  'twitter',
+  'facebook',
+];
+
+type ContactSelectorProps = {
+  control: Control<z.infer<typeof editClubSchema>>;
+  register: UseFormRegister<z.infer<typeof editClubSchema>>;
+  errors: FieldErrors<z.infer<typeof editClubSchema>>;
+  modifyDeleted: Dispatch<modifyDeletedAction>;
+};
+const ContactSelector = ({
+  control,
+  register,
+  errors,
+  modifyDeleted,
+}: ContactSelectorProps) => {
+  const { fields, append, remove } = useFieldArray({
+    control: control,
+    name: 'contacts',
+  });
+  const [available, dispatch] = useReducer(Reducer, startContacts);
+  const addNew = (platform: Contact['platform']) => {
+    dispatch({ type: 'remove', target: platform });
+    append({ platform: platform, url: '' });
+  };
+  const removeItem = (index: number, platform: Contact['platform']) => {
+    if (fields[index]?.clubId) {
```

Comment:
Try just extracting this to a varaible like 
`const field = fields[index]`
then do the check `field?.clubId`
Typescript moment :)